app/components/CategoryOverviewView.js

```diff
@@ -0,0 +1,21 @@
+import React, { Children, PropTypes, cloneElement } from 'react';
+import { View } from 'react-native';
+
+/*
+ *Passing category to all children as prop
+ */
+const CategoryOverviewView = ({ category, children }) => (
+  <View>
+    {Children.map(children, child => cloneElement(child, { category }))}
```

Comment:
My intent was to make platform-independent component which can be used to retrieve current category (and products etc) and pass it as props to platform-specific components (header, list, footer). 
I don't particularly like my solution either. I'll rework it simpler.